brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java

```diff
@@ -0,0 +1,111 @@
+package com.github.kristofa.brave;
+
+import com.github.kristofa.brave.internal.Nullable;
+import com.twitter.zipkin.gen.Span;
+import java.io.Closeable;
+import java.io.Flushable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Extend this class to offload the task of reporting spans to separate thread. By doing so, callers
+ * are protected from latency or exceptions possible when exporting spans out of process.
+ */
+public abstract class FlushingSpanCollector implements SpanCollector, Flushable, Closeable {
```

Comment:
>
> Will it break anything in the public API?
>
Nope. main reason for making this a superclass of AbstractCollector was to
prevent that.